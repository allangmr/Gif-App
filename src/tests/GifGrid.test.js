import { shallow } from "enzyme";
import GifGrid from "../components/GifGrid";
import '@testing-library/jest-dom';
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock('../hooks/useFetchGifs');

describe('Test of GifGrid.js', () => {
    const title  = 'Spiderman';
    test('should be a component of GifGrid', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        })
        const wrapper = shallow(<GifGrid category={title} />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should be show items when loading the images',() => {
        const gifs = [{
            id:15,
            url:'https://google.com',
            title: 'Cualquier Cosa'
        },
        {
            id:16,
            url:'https://google.com',
            title: 'Cualquier Cosa'
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        })
        const wrapper = shallow(<GifGrid category={title} />);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})
