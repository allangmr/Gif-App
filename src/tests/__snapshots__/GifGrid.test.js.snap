// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of GifGrid.js should be a component of GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInLeft"
  >
    Spiderman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
